sopel/modules/safety.py

```diff
@@ -99,6 +100,13 @@ def setup(bot):
                 malware_domains.add(clean_line)
 
 
+def shutdown(bot):
+    try:
+        del bot.memory['safety_cache']
+    except KeyError:
+        pass
```

Comment:
Probably OK to clear this I guess. It shouldn't cause that much repeated work.

After tracing through how the cache works though, either I don't understand `_clean_cache()` at all or it's almost completely useless and doesn't do what it should. Feel free to look at that if you get biologist's block again. 😉

Reply:
> After tracing through how the cache works though, either I don't understand _clean_cache() at all or it's almost completely useless and doesn't do what it should.

Seems like the latter. If I understand it correctly, every 24 hours (or if there are > 1024 entries), _one_ (the oldest) is removed... Not really a `_clean`'ing if you ask me :stuck_out_tongue_closed_eyes:

Reply:
That's the conclusion I came to, also. Going to make a new issue for this one.